modules/mobile/app/models/mobile/v0/adapters/lighthouse_direct_deposit_error.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    module Adapters
+      class LighthouseDirectDepositError
+        def self.parse(response)
+          body = parse_body(response[:body])
+          detail = parse_detail(body)
+          status = parse_status(response[:status], detail)
+
+          errors = [
+            {
+              title: parse_title(body),
+              detail:,
+              code: parse_code(detail),
+              source: data_source
+            }
+          ]
+
+          Lighthouse::DirectDeposit::ErrorResponse.new(status, errors)
+        end
+
+        def self.parse_status(status, detail)
+          return '500' if detail.include?('accountRoutingNumber.invalidCheckSum') ||
+                          detail.include?('payment.accountRoutingNumber.invalid') ||
+                          detail.include?('Routing number related to potential fraud')
+
+          status
+        end
+
+        def self.parse_title(body)
+          body[:error] || body[:title] || status_message_from(body[:status]) || 'Unknown error'
+        end
+
+        def self.parse_body(body)
+          body.to_hash.with_indifferent_access
+        end
+
+        def self.parse_detail(body)
+          messages = []
+          messages.push(body[:error_codes][0][:error_code]) if body[:error_codes].present?
+          messages.push(body[:error_codes][0][:detail]) if body[:error_codes].present?
+          messages.push(body[:error_description] || body[:error] || body[:detail] || body[:message] || 'Unknown error')
+
+          messages.compact.join(': ')
+        end
+
+        # rubocop:disable Metrics/MethodLength
```

Comment:
Looking at `Lighthouse::DirectDeposit::ErrorResponse` it looks like we can just disable on the same line as the method definition and then we don't have to re-enable after. Handy.